### PR TITLE
Dev/faster unity

### PR DESCRIPTION
**Pull request by maxlep**
Disable domain reload. Assorted updates focused on decreasing extra work done on recompile supporting disabled domain reload to decrease wait time in the editor. Also includes some bugfixes and warnings cleanup.

### CODE REVIEW

File: Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        [InitializeOnLoadMethod]
+        private static void Initialize()
+        {
+            _variablesToReset = new HashSet<string> {};
+            
+            EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
+            Debug.Log($"OnProjectLoadedInEditor: ScriptableObjectInitializer");
+            
+        }
+#endif
```
Comment (maxlep) [inline Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs:27]: Is there a reason these #if statements are separated?

File: Assets/BML/ScriptableObjectCore/Scripts/Variables/Variable.cs [hunk 2]
```diff
@@ -150,14 +152,19 @@ public void SetValue(T value)
...
             if (enableLogs) Debug.Log($"Reset {this.name} (Runtime value {runtimeValue})");
             
             prevValue = runtimeValue;
             runtimeValue = defaultValue;
             this.OnUpdateDelta?.Invoke(prevValue, runtimeValue);
             this.OnUpdate?.Invoke();
+#if UNITY_EDITOR
+            string fullPath = AssetDatabase.GetAssetPath(this);
+            ScriptableObjectResetOnEnterPlaymode.RegisterVariableToReset(fullPath);
+            Debug.Log("Registering " + fullPath);
```
Comment (maxlep) [inline Assets/BML/ScriptableObjectCore/Scripts/Variables/Variable.cs:166]: Remove log (or make conditional)

File: Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        private static HashSet<string> _variablesToReset;
+        
+#if UNITY_EDITOR
+        [InitializeOnLoadMethod]
+        private static void Initialize()
+        {
+            _variablesToReset = new HashSet<string> {};
+            
+            EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
+            Debug.Log($"OnProjectLoadedInEditor: ScriptableObjectInitializer");
```
Comment (maxlep) [inline Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs:24]: Remove log (or make conditional)

File: Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            
+        }
+#endif
+#if UNITY_EDITOR
+        private static void OnPlayModeStateChanged(PlayModeStateChange state)
+        {
+            if (state == PlayModeStateChange.EnteredPlayMode)
+            {
+                Debug.Log("Resetting Scriptable Variables on Enter Playmode");
```
Comment (maxlep) [inline Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs:33]: Remove log (or make conditional)

File: Assets/BML/ScriptableObjectCore/Scripts/Variables/Variable.cs [hunk 5]
```diff
@@ -3,9 +3,11 @@
 using BML.ScriptableObjectCore.Scripts.Utils;
 using Sirenix.OdinInspector;
 using Sirenix.Serialization;
-using UnityEditor;
 using UnityEngine;
 using UnityEngine.Serialization;
+#if UNITY_EDITOR
+using UnityEditor;
```
Comment (maxlep) [inline Assets/BML/ScriptableObjectCore/Scripts/Variables/Variable.cs:9]: Does this actually need to be conditional?

File: Assets/Scripts/CaveV2/InfluenceStateData.cs [hunk 6]
```diff
@@ -16,16 +21,21 @@ public class InfluenceStateData : ScriptableObject
...
         #endregion
 
-        private void Reset()
+        public event IResettableScriptableObject.OnResetScriptableObject OnReset;
+        public void ResetScriptableObject()
         {
             _currentNodes = new Dictionary<Collider, CaveNodeData>();
             _currentNodeConnections = new Dictionary<Collider, CaveNodeConnectionData>();
             _current = new Dictionary<Collider, ICaveNodeData>();
+            Debug.Log("Reset InfluenceStateData");
```
Comment (maxlep) [inline Assets/Scripts/CaveV2/InfluenceStateData.cs:35]: Remove log (or make conditional)

File: ProjectSettings/SceneTemplateSettings.json
Comment (maxlep) [inline ProjectSettings/SceneTemplateSettings.json:1]: What is this file?

File: Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs (hunk 1 above)
Comment (djc5627) [reply Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs:27]: no they can be combined

File: Assets/BML/ScriptableObjectCore/Scripts/Variables/Variable.cs (hunk 5 above)
Comment (djc5627) [reply Assets/BML/ScriptableObjectCore/Scripts/Variables/Variable.cs:9]: good question, honestly not sure

File: ProjectSettings/SceneTemplateSettings.json
Comment (djc5627) [reply ProjectSettings/SceneTemplateSettings.json:1]: No idea LOL

**Pull request merged**